readme.md

```diff
@@ -0,0 +1,109 @@
+# Heksher
+Heksher is a service that makes shared context variables a breeze. It is designed with setting management in mind, but
+can be used for any data that needs to be shared and updated throughout an entire architecture.
+
+## How does it work?
+Heksher stores a collection of settings, these are the context variables. For each setting, there is a set of rules,
+each rule has a value and a set of conditions. These conditions are set by each context inside process individually.
+So that whenever a context needs to know the value to use for each setting, it can just filter through the rules. 
+
+### Example
+Assume we have 3 context features: `account`, `user`, and `theme`. We also have a setting called `cache_size` of type
+`int`. We might have a ruleset that looks like this (conceptually, the data is stored differently):
+
+|account|user|theme|value
+|:---:|:---:|:---:|:---:
+|john|*|*|100
+|jim|*|*|50
+|jim|admin|*|200
+|*|guest|*|10
+|*|guest|dark|20
+
+Note that the rules are prioritized by last-feature first, and wildcards are prioritized last, so this rule set
+evaluates to:
+1. If the user is `guest`, using the `dark` theme, return 20.
+1. Otherwise, if the user is `guest` (without the `dark` theme, since in that case we will have returned 20 already),
+  return 10.
+1. Otherwise, if the account is `jim`, with the `admin` user, return 200.
+1. Otherwise, if the account is `jim`, return 50.
+1. Otherwise, if the account is `john`, return 100.
+1. Otherwise, return the client-specific default value (defined when a client declares the setting)
+
+Now if one of our servers is handling a request from a specific user, using a specific account, with a specific theme,
+we can find out the value of `cache_size` by simply querying the Heksher service with those context feature values.
+(it is recommended to query and cache the results in advance, see below).
+
+## How do I make it work?
+Currently, Heksher supports two environment variables, both of which are required:
+* `HEKSHER_DB_CONNECTION_STRING`: An [SqlAlchemy-style connection string](https://docs.sqlalchemy.org/en/14/core/engines.html#database-urls)
+ that leads to a postgreSql database that has been initialized (Heksher supports alembic to initialize a database).
+* `HEKSHER_STARTUP_CONTEXT_FEATURES`: A semicolon-delimited list of context features. Heksher will adapt the database's
+  existing context features to this list (or raise an error if it cannot).
+
+The service itself can be run from the docker image `<TBD>`.
```

Comment:
update this

Reply:
It will be updated once the first image is released